plugins/pay.c

```diff
@@ -1387,9 +1387,15 @@ static struct command_result *json_pay(struct command *cmd,
 static void utc_timestring(const struct timeabs *time, char str[UTC_TIMELEN])
 {
 	char iso8601_msec_fmt[sizeof("YYYY-mm-ddTHH:MM:SS.%03dZ")];
+	struct tm *t = gmtime(&time->ts.tv_sec);
```

Comment:
Interesting take, I was assuming that this could not possibly return `NULL`, btu the man page says:

> It may return NULL when the year does not fit into an integer

My best guess was this https://github.com/ElementsProject/lightning/tree/issue-4991 assuming that the buffer was too small, but yours seems more likely.

The root cause seems uninitialized memory, will dig further :-)